datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -689,6 +690,16 @@ pub fn build_join_schema(
             left_fields().chain(right_field).unzip()
         }
         JoinType::RightSemi | JoinType::RightAnti => right_fields().unzip(),
+        JoinType::RightMark => {
+            let left_field = once((
+                Field::new("mark", arrow_schema::DataType::Boolean, false),
+                ColumnIndex {
+                    index: 0, // 'mark' is not associated with either side
+                    side: JoinSide::None,
+                },
+            ));
```

Comment:
Maybe this could also be moved to a function that is used for both `LeftMark` and `RightMark`.